comps/guardrails/pii_detection/pii/pii_utils.py

```diff
@@ -71,7 +73,26 @@ def detect_pii(self, text):
 
 class PIIDetectorWithML(PIIDetector):
     def __init__(self):
+        import joblib
+        from huggingface_hub import hf_hub_download
+        from sentence_transformers import SentenceTransformer
+
         super().__init__()
+        print("Loading embedding model...")
+        embed_model_id = "nomic-ai/nomic-embed-text-v1"
+        self.model = SentenceTransformer(model_name_or_path=embed_model_id, trust_remote_code=True)
```

Comment:
the 'model_name_or_path' should be available for both local model path and remote huggingface model id. We should not assume the user must load the model from HF.

Reply:
Right now the ML strategy has only been tested with nomic embedding model + Intel trained logistic classifier. If user changed embedding model the results may be bad. So I hard coded the embedding model id.

Reply:
what i mean is that some users may have some proxy issue and cannot visit HF website from the server side. We need to leave a port for these users to load a model from the local side. If you think only the mentioned model can achieve a satisfying performance, you'd better mentioned in the readme to give a model recommendation.

Reply:
I have added a notice and disclaimer in readme. The agreement with legal is that this is only to be used for demo purpose.